src/parsing.rs

```diff
@@ -252,57 +244,58 @@ fn shorthand_hourly(i: &str) -> IResult<&str, ScheduleFields> {
         DaysOfWeek::all(),
         Years::all(),
     );
-    Ok((i, fields))
+    Ok(fields)
 }
 
-fn shorthand(i: &str) -> IResult<&str, ScheduleFields> {
+fn shorthand(i: &mut &str) -> PResult<ScheduleFields> {
     let keywords = alt((
         shorthand_yearly,
         shorthand_monthly,
         shorthand_weekly,
         shorthand_daily,
         shorthand_hourly,
     ));
-    delimited(multispace0, keywords, multispace0)(i)
-}
-
-fn longhand(i: &str) -> IResult<&str, ScheduleFields> {
-    let seconds = map_res(field, Seconds::from_field);
-    let minutes = map_res(field, Minutes::from_field);
-    let hours = map_res(field, Hours::from_field);
-    let days_of_month = map_res(field_with_any, DaysOfMonth::from_field);
-    let months = map_res(field, Months::from_field);
-    let days_of_week = map_res(field_with_any, DaysOfWeek::from_field);
-    let years = opt(map_res(field, Years::from_field));
-    let fields = tuple((
+    delimited(multispace0, keywords, multispace0).parse_next(i)
+}
+
+fn longhand(i: &mut &str) -> PResult<ScheduleFields> {
+    let seconds = field.try_map(Seconds::from_field);
+    let minutes = field.try_map(Minutes::from_field);
+    let hours = field.try_map(Hours::from_field);
+    let days_of_month = field_with_any.try_map(DaysOfMonth::from_field);
+    let months = field.try_map(Months::from_field);
+    let days_of_week = field_with_any.try_map(DaysOfWeek::from_field);
+    let years = opt(field.try_map(Years::from_field));
+    let fields = (
         seconds,
         minutes,
         hours,
         days_of_month,
         months,
         days_of_week,
         years,
-    ));
+    );
 
-    map(
-        terminated(fields, eof),
-        |(seconds, minutes, hours, days_of_month, months, days_of_week, years)| {
-            let years = years.unwrap_or_else(Years::all);
-            ScheduleFields::new(
-                seconds,
-                minutes,
-                hours,
-                days_of_month,
-                months,
-                days_of_week,
-                years,
-            )
-        },
-    )(i)
+    terminated(fields, eof)
+        .map(
+            |(seconds, minutes, hours, days_of_month, months, days_of_week, years)| {
+                let years = years.unwrap_or_else(Years::all);
+                ScheduleFields::new(
+                    seconds,
+                    minutes,
+                    hours,
+                    days_of_month,
+                    months,
+                    days_of_week,
+                    years,
+                )
+            },
+        )
+        .parse_next(i)
 }
 
-fn schedule(i: &str) -> IResult<&str, ScheduleFields> {
-    all_consuming(alt((shorthand, longhand)))(i)
```

Comment:
The lack of `all_consuming` might stand out but that is because we call `schedule.parse()` instead of `schedule.parse_next()`.  `Parser::parse` has an `eof` check built-in.